### PR TITLE
Document harvester-csi-hostname limitations

### PR DESCRIPTION
**Pull request by connorkuehl**

### CODE REVIEW

File: docs/rancher/csi-driver.md [hunk 1]
```diff
@@ -20,6 +20,9 @@ The Harvester Container Storage Interface (CSI) Driver provides a standard CSI i
 
 - The Kubernetes cluster is built on top of Harvester virtual machines.
 - The Harvester virtual machines run as guest Kubernetes nodes are in the same namespace.
```
Comment (LucasSaintarbor) [inline docs/rancher/csi-driver.md:22]:
```suggestion
- The Harvester virtual machines run as guest Kubernetes nodes in the same namespace.
```

File: versioned_docs/version-v1.1/rancher/csi-driver.md [hunk 2]
```diff
@@ -20,6 +20,9 @@ The Harvester Container Storage Interface (CSI) Driver provides a standard CSI i
 
 - The Kubernetes cluster is built on top of Harvester virtual machines.
 - The Harvester virtual machines run as guest Kubernetes nodes are in the same namespace.
```
Comment (LucasSaintarbor) [inline versioned_docs/version-v1.1/rancher/csi-driver.md:22]:
```suggestion
- The Harvester virtual machines run as guest Kubernetes nodes in the same namespace.
```

File: docs/rancher/csi-driver.md [hunk 3]
```diff
@@ -20,6 +20,9 @@ The Harvester Container Storage Interface (CSI) Driver provides a standard CSI i
...
+- The Harvester virtual machine guests' hostnames match their corresponding Harvester virtual machine names.
+  We hope [to remove this limitation](https://github.com/harvester/harvester/issues/4396) in a future release
+  of Harvester.
```
Comment (LucasSaintarbor) [inline docs/rancher/csi-driver.md:25]:
```suggestion
- The Harvester virtual machine guests' hostnames match their corresponding Harvester virtual machine names. Guest cluster Harvester VMs can't have different hostnames than their Harvester VM names when using the Harvester CSI driver. We hope [to remove this limitation](https://github.com/harvester/harvester/issues/4396) in a future release of Harvester.
```

File: versioned_docs/version-v1.1/rancher/csi-driver.md [hunk 4]
```diff
@@ -20,6 +20,9 @@ The Harvester Container Storage Interface (CSI) Driver provides a standard CSI i
...
+- The Harvester virtual machine guests' hostnames match their corresponding Harvester virtual machine names.
+  We hope [to remove this limitation](https://github.com/harvester/harvester/issues/4396) in a future release
+  of Harvester.
```
Comment (LucasSaintarbor) [inline versioned_docs/version-v1.1/rancher/csi-driver.md:25]:
```suggestion
- The Harvester virtual machine guests' hostnames match their corresponding Harvester virtual machine names. Guest cluster Harvester VMs can't have different hostnames than their Harvester VM names when using the Harvester CSI driver. We hope [to remove this limitation](https://github.com/harvester/harvester/issues/4396) in a future release of Harvester.
```

**Review by LucasSaintarbor — CHANGES_REQUESTED** [review]
Lmk if these changes are okay, thanks!

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Review by LucasSaintarbor — APPROVED** [review]
LGTM, thanks!

File: docs/rancher/csi-driver.md (hunk 1 above)
Comment (LucasSaintarbor) [reply docs/rancher/csi-driver.md:22]: @connorkuehl Makes sense, thank you!

**Review by vickyhella — APPROVED** [review]
LGTM!

**Pull request merged**